### PR TITLE
Added `allowMultiline` property to the `jsx-curly-spacing` rule

### PR DESCRIPTION
**Pull request by mathieumg**
What do you think about this? I had hoped it would be like that by default when I first read about that rule being introduced! Which prompts my next question: should `allowMultiline` be on by default? I currently put it off by default and I'm ignoring it when the option is set to `"always"` as to not break current users (new warnings/errors) that do:

```js
<Hello name={
  firstname
} />;
```

using

```json
"jsx-curly-spacing": [2, "always"]
```

If it were on by default, I could leverage the option in all cases, meaning users would have a way to enforce:

```js
// Valid
<Hello name={ firstname } />;

// Invalid
<Hello name={
  firstname
} />;
```

with

```json
"jsx-curly-spacing": [2, "always", {"allowMultiline": false}]
```

However, current users relying on

```js
<Hello name={
  firstname
} />;
```

to warn with the current defaults, would not be warned anymore.

### CODE REVIEW

**Comment by yannickcr** [comment]
Thanks for this!

I think it's ok to make a small break on this rule to make it more flexible.

**Comment by mathieumg** [comment]
So, which break is it? :stuck_out_tongue:  Default to `true`, but rule also considered with `"always"` if set to `false`? (Last example I showed)

If so, I'll update the PR!

**Comment by moll** [comment]
:+1: This would then match http://eslint.org/docs/rules/object-curly-spacing.html which allows multilines in the `never` setting.

**Comment by yannickcr** [comment]
> So, which break is it? :stuck_out_tongue: Default to true, but rule also considered with "always" if set to false? (Last example I showed)

Yep, that break ;)

File: lib/rules/jsx-curly-spacing.js [hunk 1]
```diff
@@ -79,16 +112,28 @@ module.exports = function(context) {
    * @returns {void}
    */
   function validateBraceSpacing(node, first, second, penultimate, last) {
```
Comment (mathieumg) [inline lib/rules/jsx-curly-spacing.js]: Refactored this part a bit because it was getting hard to follow! I don't know what your stance is on early returns to avoid deep nesting, but I can change this to an else statement if that is what you prefer.

File: lib/rules/jsx-curly-spacing.js [hunk 2]
```diff
@@ -10,13 +10,24 @@
 
 module.exports = function(context) {
   var spaced = context.options[0] === 'always';
+  var multiline = context.options[1] ? context.options[1].allowMultiline : true;
```
Comment (mathieumg) [inline lib/rules/jsx-curly-spacing.js]: This is a little fragile and relies on the fact that it's the only option and that no one would pass `{}` as the second parameter, I can change it to leverage `hasOwnProperty` if you want.

Comment (yannickcr) [inline lib/rules/jsx-curly-spacing.js]: Should be ok since `allowMultiline` is the only option authorized by the rule schema.

File: lib/rules/jsx-curly-spacing.js (hunk 1 above)
Comment (yannickcr) [inline lib/rules/jsx-curly-spacing.js]: LGTM

File: lib/rules/jsx-curly-spacing.js (hunk 2 above)
Comment (mathieumg) [inline lib/rules/jsx-curly-spacing.js]: Yeah didn't think about the schema, good!
